nimble/host/src/ble_hs_mbuf.c

```diff
@@ -84,6 +98,50 @@ ble_hs_mbuf_l2cap_pkt(void)
     return ble_hs_mbuf_gen_pkt(BLE_HCI_DATA_HDR_SZ + BLE_L2CAP_HDR_SZ);
 }
 
+#if MYNEWT_VAL(BLE_HS_OWN_MBUF_FOR_HEADERS)
+/**
+ * Allocates an mbuf for use by the nimble host.
+ */
+static struct os_mbuf *
+ble_hs_mbuf_gen_hdr_pkt(uint16_t leading_space)
+{
+    struct os_mbuf *om;
+    int rc;
+
+    om = os_mbuf_get_pkthdr(&ble_hs_hdr_mbuf_pool, leading_space);
+    if (om == NULL) {
+        return NULL;
+    }
+
+    if (om->om_omp->omp_databuf_len < leading_space) {
+        rc = os_mbuf_free_chain(om);
```

Comment:
Given that this is a dedicated pool should we presume that we always have the proper space and simply assert() here if the asked for leading space is not enough?

Reply:
agree